automation_tools/__init__.py

```diff
@@ -1622,6 +1623,30 @@ def setenforce(mode):
     run('setenforce {0}'.format(mode))
 
 
+def host_pings(host):
+    """This ensures the given IP/hostname pings succesfully.
+
+    :param host: A string. The IP or hostname of host.
+
+    """
+    if host is None:
+        print("The hostname/IP is not provided to ping. \
+        Terminating the execution !!")
+        sys.exit(1)
+    while True:
+        command = subprocess.Popen(
+            'ping -c1 {0}'.format(host),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True
+        )
+        output = command.communicate()[0]
+        if '1 received' in output:
```

Comment:
How about using return codes instead?

```
       If ping does not receive any reply packets at all it will exit with  code  1.  If  a
       packet  count  [ping(8)]  and  deadline [ping(8)] are both specified, and fewer than
       count [ping(8)] packets are received by the time the deadline [ping(8)] has arrived,
       it  will  also  exit with code 1.  On other error it exits with code 2. Otherwise it
       exits with code 0. This makes it possible to use the exit code to see if a  host  is
       alive or not.
```